### ISSUE TITLE
BUG: shap library

### ISSUE THREAD

**Issue by Mfundo-debug** [shap/shap#3052]
### Issue Description

there's an issue within the shap library
1. created new virtual env and created a specific dir
2.  `conda install ipykernel`
3. `conda install -c conda-forge shap` i also tried `pip install shap` but i still errors
below are the screenshots i took during the process @thatlittleboy 

" NumbaDeprecationWarning: The 'nopython' keyword argument was not supplied to the 'numba.jit' decorator. The implicit default value for this argument is currently False, but it will be changed to True in Numba 0.59.0. See https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit for details."

![installation_1](https://github.com/slundberg/shap/assets/71297224/234f52c1-7908-427c-a85b-c1c47df05481)
![installation_2](https://github.com/slundberg/shap/assets/71297224/cab7f481-008e-42a4-a9c6-3cb7d3ae47a6)
![installation_3](https://github.com/slundberg/shap/assets/71297224/99cbd47a-91d7-4aaf-bf28-a74e87f198a2)
![installation_4](https://github.com/slundberg/shap/assets/71297224/da18d8ad-65f9-4e88-9dfd-439e2e352632)
![installation_5](https://github.com/slundberg/shap/assets/71297224/b55382ff-f36c-48f0-aad9-e9b41214aa47)

### Minimal Reproducible Example

```python
import xgboost
import shap

#train an XGBoost model
X,y = shap.datasets.adult()
model = xgboost.XGBRegressor().fit(X, y)
#compute SHAP values
explainer = shap.Explainer(model)
shap_values = explainer(X)

# visualize the first prediction's explanation
shap.plots.waterfall(shap_values[0])
```

### Expected Behavior

expected outputs

### Bug report checklist

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest release](https://github.com/slundberg/shap/releases) of shap.
- [ ] I have confirmed this bug exists on the [master branch](https://github.com/slundberg/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [ ] I'd be interested in making a PR to fix this bug

### Installed Versions

0.41.0

**Comment by thatlittleboy** [comment]
Hi @Mfundo-debug , are you able to try installing from the master branch via pip? https://github.com/slundberg/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch

**Comment by Mfundo-debug** [comment]
I tried that and this is what i got
![error](https://github.com/slundberg/shap/assets/71297224/f9f1d6dc-f15c-4d70-9b40-169dc8d379d3)
![error_1](https://github.com/slundberg/shap/assets/71297224/767c4f57-40b5-4023-9667-67ad184e7ea8)

![error_2](https://github.com/slundberg/shap/assets/71297224/e912a191-1e25-4617-860e-ece892ba1d4f)

**Comment by connortann** [comment]
This issues was resolved by #2943. You can pip install these changes from the git repo or wait for the next PyPI release.
